registry/storage/driver/s3-aws/s3.go

```diff
@@ -1300,12 +1355,12 @@ func (w *writer) Write(p []byte) (int, error) {
 	// If the last written part is smaller than minChunkSize, we need to make a
 	// new multipart upload :sadface:
 	if len(w.parts) > 0 && int(*w.parts[len(w.parts)-1].Size) < minChunkSize {
-		var completedUploadedParts completedParts
-		for _, part := range w.parts {
-			completedUploadedParts = append(completedUploadedParts, &s3.CompletedPart{
+		completedUploadedParts := make(completedParts, len(w.parts))
+		for i, part := range w.parts {
+			completedUploadedParts[i] = &s3.CompletedPart{
 				ETag:       part.ETag,
 				PartNumber: part.PartNumber,
-			})
+			}
 		}
 
 		sort.Sort(completedUploadedParts)
```

Comment:
Unrelated: wondering if `sort.Stable` (or `slices.SortStableFunc` as recommended on that function) would bring performance benefits.

Reply:
No, `sort.Stable` is slower than `sort.Sort`.
> Sort sorts data in ascending order as determined by the Less method. It makes one call to data.Len to determine n and **O(n*log(n)) calls to data.Less and data.Swap.** The sort is not guaranteed to be stable.

> [Stable] makes one call to data.Len to determine n, <strong>O(n\*log(n)) calls to data.Less and O(n*log(n)*log(n)) calls to data.Swap.</strong>

Additionally, the two sort algorithms will produce identical results as every `PartNumber` is unique in the slice.

Reply:
Looks like I had my wires crossed there, and somehow misremembered  that `Stable` did less (so faster) than `Sort`; thanks!